test/f90_correct/src/implied_do10.f90

```diff
@@ -0,0 +1,43 @@
+!
+! Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+! See https://llvm.org/LICENSE.txt for license information.
+! SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+!
+! Test for implied-do loop containing function call that returns adjustable
+! array (https://github.com/flang-compiler/flang/issues/1253).
+
+program main
+  implicit none
+  integer :: x(2) = 0, y(3), z(18)
+  integer :: i, j
+  integer :: cmp(18) = [11, 21, 21, 12, 12, 22, 22, 22, 22, &
+                        13, 13, 13, 23, 23, 23, 23, 23, 23]
+
+  x = (/ (f(1), i = 1, 2) /)
+  if (any(x /= 4)) STOP 1
+
```

Comment:
As I understand, this is an extended version of the `implied_do10.f90` test case from PR #1254. The version introduced there was so small and simple for a reason: this `implied do` operation performs an allocation of a small temporary space by calling `f90_alloc04a_i8()` function, which is intercepted by a replacement stub function presented in the `implied_do10.c` file. But keep in mind that this replacement function ends either with `abort()` or with `exit(0)` but it never returns. This was intentional: what we are testing here is that the number of allocated elements was actually computed and not picked randomly (due to a faulty code). Hence the rest of this test case has no use. Also it is up to the stub replacement to print `"PASSED"`, the `print *, "PASS"` below is overhead.

Reply:
Thanks for your comments. I've changed the test case to the one in PR #1254 to keep it small and simple.  Please notice that I modified
`if ((!(*nelem)) || (2UL < *nelem))`  to  `if (2UL < *nelem)`, because `nelem == 0` is expected at this time.